sdk/core/core/inc/az_http.h

```diff
@@ -12,7 +12,7 @@
 #include <_az_cfg_prefix.h>
 
 enum {
-  AZ_HTTP_URL_MAX_SIZE = 1024 * 2,
+  AZ_HTTP_URL_MAX_SIZE = 1024,
```

Comment:
storage urls can be quite long.  Is there a reason we need to shrink the size by default

Reply:
I reduce it because now we need to allocate this 2 times in stack.  
one time inside any sdk client (keyvault, storage) as immutable and a second time on each API (create, download, delete, etc) where we copy the immutable url from client to this second buffer so we can now add path/queryParameter/etc.
So, we are still using 2k, but 1k for initial url and then another 1k to actually use it

This is at the end to allow concurrency for multiple threads using one client

Reply:
Shrinking the size because we need 2 of them isn't a good reason. We are just trying 1024 for now and, if we have t increase it, we'll increase it. I think KeyVault will be OK with 1024; It's Storage that worries me. That being said, we need some kind of .h file that has all these enums in it so that customers can easily adjust them and re-compile. Only the customer really knows how big their urls will become and so they must have the ultimate control.